libs/pilight/core/datetime.c

```diff
@@ -612,22 +612,26 @@ time_t datetime2ts(int year, int month, int day, int hour, int minutes, int seco
 int tzoffset(char *tz1, char *tz2) {
 	logprintf(LOG_STACK, "%s(...)", __FUNCTION__);
 
+	if (strcmp(tz1, tz2) == 0) {	// prevent heavy calculations on simple case
+		return 0;
+	}
+
 	atomiclock();
 
 	time_t utc, tzsearch, now;
 	struct tm tm;
-	char oritz[64];
+	char oritz[64], s;
 
 	memset(&tm, '\0', sizeof(struct tm));
 	memset(oritz, '\0', 64);
 
-	if(getenv("TZ") != NULL) {
-		strcpy(oritz, getenv("TZ"));
+	if((s = getenv("TZ")) != NULL) {
+		strcpy(oritz, s);
 	}
 
 	now = time(NULL);
 #ifdef _WIN32
-	localtime(&now);
+	tm = *localtime(&now);
```

Comment:
I didn't check if the output is used. But in the #else case (e.g. _WIN32 not defined) the variable tm is being set with the localtime result whether or not tm is used later. If tm is not used, the code setting it up could be removed et all. You you want it this way?